File: src/app/(api)/api/dashboard/DashboardResponse.ts
import { PageResponse } from "./PageResponse";
import { ProfileResponse } from "./ProfileResponse";

export class DashboardResponse {
  profile: ProfileResponse;
  pages: PageResponse;

  constructor(profile: ProfileResponse, pages: PageResponse) {
    this.profile = profile;
    this.pages = pages;
  }
}

File: aml-backend/src/main/java/dev/abarmin/aml/dashboard/rest/response/DashboardResponse.java
package dev.abarmin.aml.dashboard.rest.response;

import dev.abarmin.aml.dashboard.model.BlockTypeModel;
import dev.abarmin.aml.dashboard.model.DashboardProfile;

import java.util.Collection;

public record DashboardResponse(
  DashboardProfile profile,
  Collection<BlockTypeModel> availableBlocks,
  PageResponse currentPage
) {
}
